### ISSUE TITLE
Filters: models with multiple foreignkey to same model override

### ISSUE THREAD

**Issue by buxx** [flask-admin/flask-admin#1250]
Hi,

This issue is related to  #909. If my admin view set column filters, but them columns are relationship on same model, last column override previous.

This code expose the bug (`Company.members` and `Company.contacts` are relations on same model):

```
# -*- coding: utf-8 -*-
from flask import Flask
from flask.ext.admin import Admin
from flask.ext.admin.contrib.sqla import ModelView
from flask_sqlalchemy import SQLAlchemy

app = Flask(__name__)
app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///'
db = SQLAlchemy(app)


class User(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    username = db.Column(db.String(128), unique=True)

    company_member_id = db.Column(db.Integer(), db.ForeignKey('company.id'))
    company_contact_id = db.Column(db.Integer(), db.ForeignKey('company.id'))

    company_member = db.relationship("Company", foreign_keys=[company_member_id], back_populates="members")
    company_contact = db.relationship("Company", foreign_keys=[company_contact_id], back_populates="contacts")


class Company(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String(128), unique=True)

    members = db.relationship("User", foreign_keys="[User.company_member_id]", back_populates="company_member")
    contacts = db.relationship("User", foreign_keys="[User.company_contact_id]", back_populates="company_contact")


class CompanyView(ModelView):
    column_filters = ("members.username", "contacts.username")

admin = Admin(app)
admin.add_view(CompanyView(Company, session=db.session))


if __name__ == '__main__':
    db.create_all()
    app.run()
```

Result is:
![selection_003](https://cloud.githubusercontent.com/assets/1104637/14249513/612ba38c-fa7a-11e5-9621-bb9a805f6228.png)

If i do filter with admin UI `User / Username` field, join will be on Company.contacts. It can be fixed ? 

Thank's !

**Comment by buxx** [comment]
I just manipulate the code to make working but dirty version: https://github.com/flask-admin/flask-admin/compare/master...algoo:dev/1.4.0/joins

This diff illustrate how fix the problem. It should be very appreciated what you implement this correctly (i do it without know entire mechanisms !) :)

**Comment by xqliu** [comment]
Dear @buxx 

Could you please create a pull request for this fix to flask-admin/flask-admin ? 

I think this fix is quite valuable as I am also plaguing by this issue.

Thanks.

**Comment by buxx** [comment]
@xqliu Hi, i make a PR #1281.

**Comment by xqliu** [comment]
@buxx That's really helpful!  thanks for the work.

**Comment by petrus-jvrensburg** [comment]
fixed in #1281
